api/_hyperfunctions/time_bucket_gapfill/time_bucket_gapfill.md

```diff
@@ -0,0 +1,69 @@
+---
+api_name: time_bucket_gapfill()
+excerpt: Bucket rows by time interval while filling gaps in data
+topics: [hyperfunctions]
+api:
+  license: community
+  type: function
+  version:
+    stable: 1.2.0
+hyperfunction:
+  family: gapfilling
+  type: bucket
+  bucket_function: time_bucket_gapfill()
+api_details:
+  summary: >
+    Group data into buckets based on time interval, while filling in gaps of missing data.
+    If you don't provide a gapfilling algorithm, such as `locf` or `interpolate`, gaps are left as `NULL` in the returned data.
+  signatures:
+    - language: sql
+      code: |
+        time_bucket_gapfill(
+          bucket_width INTERVAL | INTEGER,
+          time TIMESTAMPTZ | INTEGER,
+          [, start TIMESTAMPTZ | INTEGER]
+          [, finish TIMESTAMPTZ | INTEGER]
+          [, timezone TEXT]
+        ) RETURNS TIMESTAMPTZ
+  parameters:
+    required:
+      - name: bucket_width
+        type: INTERVAL | INTEGER
+        description: >
+          A PostgreSQL time interval to specify the length of each bucket.
+          For example, use `1 day` to get daily buckets.
+          Use `INTEGER` only if your time column is integer-based.
+      - name: time
+        type: TIMESTAMPTZ | INTEGER
+        description: The timestamp on which to base the bucket
+    optional:
+      - name: start
+        type: TIMESTAMPTZ | INTEGER
+        description: >
+          The start of the period to gapfill.
+          Values before `start` are passed through, but no gapfilling is performed.
+          Use `INTEGER` only if your time column is integer-based.
+          In TimescaleDB 1.3, we recommend using a `WHERE` clause rather than specifying `start`.
+          The `WHERE` is more performant, because the query planner can filter out chunks by constraint exclusion.
```

Comment:
```suggestion
          We recommend using a `WHERE` clause rather than specifying `start`, which is a legacy option.
          The `WHERE` is more performant, because the query planner can filter out chunks by constraint exclusion.
```